### PR TITLE
embed typed DBChainSet and DBNodeSet interfaces in ChainSet

### PR DESCRIPTION
**Pull request by cfal**

### CODE REVIEW

File: core/chains/evm/chain_set.go [hunk 1]
```diff
@@ -40,29 +40,20 @@ type ChainConfigUpdater func(*types.ChainCfg) error
 //go:generate mockery --quiet --name ChainSet --output ./mocks/ --case=underscore
 type ChainSet interface {
 	services.ServiceCtx
-	Get(id *big.Int) (Chain, error)
+	chains.DBChainSet[utils.Big, *types.ChainCfg]
```
Comment (augustbleeds) [inline core/chains/evm/chain_set.go:43]: am i following this correctly: the chain set for ethereum would include goerli, seopolia, mainnet, etc. ? it's a class that encompasses all deployed versions of a blockchain?

**Review by augustbleeds — APPROVED** [review]
awesome, looks nice and clean now :)

File: core/chains/evm/chain_set.go (hunk 1 above)
Comment (jmank88) [reply core/chains/evm/chain_set.go:43]: In practice, yes, each ChainSet holds all the Chains for one "family", which for EVM includes all those variants.
This all becomes much simpler after https://github.com/smartcontractkit/chainlink/pull/8434

Comment (samsondav) [reply core/chains/evm/chain_set.go:43]: Yes that is correct. Plus polygon, fantom, all evm chains

**Review by samsondav — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

**Pull request merged**